src/Network.cpp

```diff
@@ -860,9 +861,40 @@ Network::Netresult Network::get_scored_moves(
     if (ensemble == DIRECT) {
         assert(rotation >= 0 && rotation <= 7);
         result = get_scored_moves_internal(state, planes, rotation);
+    } else if (ensemble == MULTI_AVG && rotation > 1) {
+        // MULTI_AVG rotation is the # of random rotations to provide
+        assert(rotation >= 1 && rotation <= 8);
+        std::vector<int> r_list{0, 1, 2, 3, 4, 5, 6, 7};
```

Comment:
Not clear why this is needed?

Reply:
If we're going to average 2-7 rotations we need to choose them randomly without replacement. I thought the simple way to handle that was shuffle a list of index numbers 0-7 then walk that list in order for the # of rotations requested. Is there a better way I overlooked? I didn't want 3 rotations to always be 0,1,2.

Reply:
Although I did always assume 'heatmap' by itself choose a random one that stuck around, and wasn't tied to 'heatmap 0'.

Reply:
Or did you mean since we're only doing rotation=8 for 'heatmap average' we don't need support for 2-7 and should delete that until someone needs it one day?

Reply:
No it just wasn't clear to me what the array was there for. Looks like std::sample is only in C++17 and it's not in boost, so it indeed seems the best way to do this.